src/socket.rs

```diff
@@ -70,12 +66,13 @@ pub struct Socket {
     /// * None - spawned task has been joined once.
     join_handle: AtomicTake<JoinHandle<Result<(), ShutdownError>>>,
 }
+
 impl Socket {
     /// Spawns a new [Socket] that must be [connect]ed.
-    pub async fn spawn(mut url: Url) -> Result<Arc<Self>, SocketError> {
+    pub async fn spawn(mut url: Url) -> Result<Arc<Self>, SpawnError> {
         match url.scheme() {
             "wss" | "ws" => (),
-            _ => return Err(SocketError::InvalidUrl(url)),
+            _ => return Err(SpawnError::InvalidUrl(url)),
```

Comment:
It might be better to make this an `UnsupportedScheme` error as this isn't actually an invalid url

Reply:
Changed